src/js/background-script.js

```diff
@@ -2,36 +2,95 @@
 const CoinformClient = require('./coinform-client');
 const CoInformLogger = require('./coinform-logger');
 
-// let browser = browser || chrome;
+const browserAPI = chrome || browser;
+
+let configuration;
 let client;
 let logger;
 
-logger = new CoInformLogger(CoInformLogger.logTypes.warning);
+let coinformUserToken = null;
+
+//Read the configuration file and if it was successful, start
+fetch(browserAPI.runtime.getURL('../resources/config.json'), {
+  mode: 'cors',
+  header: {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  }
+})
+  .then(res => res.json())
+  .then(res => {
+
+    configuration = res;
+    logger = new CoInformLogger(CoInformLogger.logTypes[configuration.coinform.logLevel]);
+    client = new CoinformClient(fetch, configuration.coinform.apiUrl);
+
+    browserAPI.runtime.onMessage.addListener(listenerRuntime);
+
+  })
+  .catch(err => {
+    console.error('Could not load configuration file', err)
+  });
+
+browserAPI.storage.local.get(['userToken'], (data) => {
+  if (data.userToken) {
+    coinformUserToken = data.userToken;
+  }
+});
+
+browserAPI.storage.onChanged.addListener(function(changes, namespace) {
+  for (let key in changes) {
+    if (key === "userToken") {
+      let storageChange = changes[key];
+      if (storageChange.newValue) {
+        if (logger) {
+          logger.logMessage(CoInformLogger.logTypes.info, `User logged in: ${storageChange.newValue}`);
+        }
+        coinformUserToken = storageChange.newValue;
+      }
+      else {
+        if (logger) {
+          logger.logMessage(CoInformLogger.logTypes.info, `User logged out: ${storageChange.oldValue}`);
+        }
+        coinformUserToken = null;
+      }
+    }
+  }
+});
 
 const listenerRuntime = function(request, sender, sendResponse) {
 
   if (request.contentScriptQuery === "ConfigureBackground") {
     
-    client = new CoinformClient(fetch, request.coinformApiUrl);
+    if (request.coinformApiUrl) {
+      logger.logMessage(CoInformLogger.logTypes.debug, `Configuriong client API url: ${request.coinformApiUrl}`);

```

Comment:
Typo in log message

Reply:
fixed in commit 0f086fb62b17c6db672674cb57de5605b635090e